server/internal/request.go

```diff
@@ -110,6 +124,47 @@ func getAdapterRequest[Config any](
 	return
 }
 
+// canAccessAdapter verifies the request has the correct token to access the
+// adapter. Will return true if the provided token matches any of the tokens
+// specified in the file located at AUTH_TOKENS_PATH.
+// Otherwise, will return false.
+func canAccessAdapter(ctx context.Context) bool {
```

Comment:
From the look of it, `GetPage` should be returning `api_adapter_v1.NewGetPageResponseError(err)` and not an actual error. I updated `validateAuthenticationToken` to return `*api_adapter_v1.Error` instead of `error` so we can return this directly

Reply:
The advantage of returning the gRPC status code is that this translates into an HTTP 401 status code in the gRPC response. That leaks less information to any potential attacker accessing this service directly than what you propose.

Reply:
Ok, reverted this to return `status.Errorf(codes.Unauthenticated, "invalid or missing token")` instead of an `api_adapter_v1.NewGetPageResponseError(err)`